hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/io/storage/row/HoodieRowDataCreateHandle.java

```diff
@@ -90,6 +91,7 @@ public HoodieRowDataCreateHandle(HoodieTable table, HoodieWriteConfig writeConfi
     this.fileId = fileId;
     this.newRecordLocation = new HoodieRecordLocation(instantTime, fileId);
     this.preserveHoodieMetadata = preserveHoodieMetadata;
+    this.skipMetadataWrite = skipMetadataWrite;
```

Comment:
Isn't the flag `preserveHoodieMetadata` already control this behavior, there is another PR raised by @usberkeley for fixing all the scenarios BTW: https://github.com/apache/hudi/pull/12404

Reply:
Yes, I saw https://github.com/apache/hudi/pull/12404 and was confused by ticket name, which mention Flink table config `hoodie.populate.meta.fields`, and didn't find any changes in `hudi-flink-client` or `hudi-flink-datasource` that will change current behavior. My point of view I described in https://github.com/apache/hudi/pull/12404#issuecomment-2550559944
And to support my comment, I created this MR that shows the lack of support of hoodie.populate.meta.fields` in Flink.

Reply:
`preserveHoodieMetadata` actually used in the code as an indicator to get metadata from row data or generate it by calling corresponding methods. And it actually a little bit confusing naming then. I suppose, values for `preserveHoodieMetadata` are defined by this schema:
![preserveHoodieMetadata](https://github.com/user-attachments/assets/55f65bf5-0fec-466d-8281-411ce767a1ca)
Looks like `preserveHoodieMetadata` could be true only for clustering operator.

Reply:
It looks like the flag `preserveHoodieMetadata` indicates whether the source row includes the metadata fields already, for table service like clustering, this should be true by default(because clustering is just a rewrite). For regular write, the metadata fields should be generated on the fly.

Let's check in which case the option `hoodie.populate.meta.fields` could be false.

Reply:
User sets value for `hoodie.populate.meta.fields` option, which is `true` by default. And in description for this config, "append only/immutable data" is mentioned as use case:
https://github.com/apache/hudi/blob/9da3221a79465f3326ae3ac206b08d60864ddcaa/hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java#L261-L265
For this reason, in this MR I supported `hoodie.populate.meta.fields` in Flink only for append mode.

Checked that everything is working for append mode by:

```SQL
CREATE TABLE hudi_debug (
    id INT,
    part INT,
    desc STRING,
    PRIMARY KEY (id) NOT ENFORCED
) 
WITH (
    'connector' = 'hudi',
    'path' = '...',
    'table.type' = 'COPY_ON_WRITE',
    'write.operation' = 'insert',
    'hoodie.populate.meta.fields' = 'false'
);
```

```SQL
INSERT INTO hudi_debug VALUES 
    (1,100,'aaa'),
    (2,200,'bbb');
```

```SQL
SELECT * FROM hudi_debug;
```

I got as a result expected:

```Text
          id        part                           desc
           2         200                            bbb
           1         100                            aaa
```

Reply:
I've also found that we could write MOR table in upsert mode without metadata, but we couldn't read it later by Flink. I've created separate bug for this, will check it: HUDI-8785

Reply:
Can we atleast add an integration test in `ITTestHoodieDataSource`

Reply:
@danny0405 
I've added `ITTestHoodieDataSource::testWriteWithoutMetaColumns`. But for proper checking, it would be great to write data by Flink, and then read it by Spark, because in Spark
```SQL
SELECT * FROM table;
```
will return all columns including those with metadata.
And it would be really useful check of engines interoperability. I've created a corresponding task HUDI-8788.